### PR TITLE
feat(tracking): add trackio as an experiment tracker alongside wandb

### LINKED ISSUE #4319: Add trackio as an experiment tracker alongside wandb
### Ticket Type

💡 Feature Request / Improvement

### Environment & System Info

```Shell

```

### Description

**Motivation:** let `lerobot-train` log to [trackio](https://github.com/gradio-app/trackio) as an alternative to wandb.

Now `wandb` is the only way to get a training curve out of `lerobot-train`. That means an account
and an API key before you can see whether your loss is going down, which is friction for a first
training run and a hard stop on machines that cannot reach wandb. 

[Trackio](https://github.com/gradio-app/trackio) needs neither: runs go to a local SQLite database you browse with `trackio show`, and if you do want to look at a run from elsewhere, the dashboard can be hosted on a Hugging Face Space.

The tracking config is currently wandb-specific (`cfg.wandb`, `--wandb.enable=true`,
`WandBLogger` constructed inline in `lerobot_train.py` and `rl/learner.py`), so a second tracker
cannot be added without either special-casing it in the training scripts or reusing fields named
after wandb.

**Proposal:** introduce a tracker choice registry, so trackers are selected the way policies,
optimizers and environments already are:

```bash
lerobot-train ... --tracker.type=wandb    # what --wandb.enable=true does today
lerobot-train ... --tracker.type=trackio  # new
```

with a small `TrackerLogger` protocol (`log_dict` / `log_policy` / `log_video`) and a
`make_tracker(cfg)` factory, so the training scripts stop knowing which backend they are talking to
and a third tracker later is a config subclass plus a logger.

Existing `--wandb.*` flags would keep working unchanged and simply build the equivalent wandb
tracker, so no existing command, config or checkpoint breaks.

**PR**: #4320 

### Context & Reproduction

**End to end in about a minute, no account and no API key** 

```bash
pip install -e ".[trackio]"

lerobot-train \
    --dataset.repo_id=lerobot/pusht \
    --policy.type=act \
    --policy.push_to_hub=false \
    --steps=200 \
    --save_checkpoint=false \
    --tracker.type=trackio \
    --tracker.project=trackio-demo

trackio show --project trackio-demo
```

Design points that came up while implementing this, and where maintainer input would help most:

1. **Should the `--wandb.*` flags be deprecated?** My patch keeps them working and silently builds
   the equivalent tracker, with no deprecation warning, because deprecating a documented flag feels
   like a call for the team rather than something to settle inside a feature PR. Happy to add the
   warning and update the remaining docs if you would rather move users over.
2. **`WandBConfig.enable` has no meaning once a tracker is selected**, since selecting one is what
   enables it (and `--tracker=null` turns tracking off again). The wandb tracker config inherits the
   field from the existing `WandBConfig` and I neutralize it rather than duplicating the field list;
   there is a cleaner shape that costs one extra dataclass, and I am happy to go either way.

### Relevant logs or stack trace

```Shell

```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [x] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

### Additional Info / Workarounds

_No response_

### PR DESCRIPTION
**Pull request by zuoxingdong**
## Summary / Motivation

**Adds [trackio](https://github.com/gradio-app/trackio) as a second experiment tracker, selected through a new `--tracker.type` registry.**

- **Motivation:** wandb is the only way to get a training curve out of `lerobot-train`, and it needs an account plus an API key. That is friction where wandb is unreachable or restricted.
- **Trackio:** no account, no API key. Runs go to a local SQLite DB you open with `trackio show`, or to a Hugging Face Space if you want to view them from elsewhere.
- **Shape:** `--tracker.type=wandb|trackio` (a draccus choice registry, like `--policy.type`), a `TrackerLogger` protocol (`log_dict` / `log_policy` / `log_video`), and a `make_tracker(cfg)` factory. The training scripts no longer name a backend, so a third tracker is a config subclass plus a logger.
- **Compatibility:** **no breaking changes.** `--wandb.*` is untouched and builds the equivalent tracker. Configs and checkpoints written before this PR load unchanged.

### Try it (about a minute, no account)

```bash
pip install -e ".[trackio]"

lerobot-train \
    --dataset.repo_id=lerobot/pusht \
    --policy.type=act \
    --policy.push_to_hub=false \
    --steps=200 \
    --save_checkpoint=false \
    --tracker.type=trackio \
    --tracker.project=trackio-demo

trackio show --project trackio-demo
```

image

## Related issues

- Fixes / Closes: #4319
- Related: # (if any)

## What changed

**New**

- **`TrackerConfig` registry** in `configs/default.py`: `WandBTrackerConfig` and `TrackioTrackerConfig`, exposed as `TrainPipelineConfig.tracker` (`None` = tracking off, `--tracker=null` to turn it off again).
- **`common/tracker_utils.py`**: the `TrackerLogger` protocol, `make_tracker`, and `cfg_to_group` (moved here from `wandb_utils`, which re-exports it, so both loggers share it).
- **`common/trackio_utils.py`**: `TrackioLogger`, mirroring `WandBLogger`. The `trackio` import is function-local, so importing lerobot is unaffected when the extra is absent.
- **`trackio` extra**, deliberately **not** in `all` (it pulls gradio, starlette, uvicorn). `uv.lock` regenerated for it: additive only, no version changes to existing packages.

**Changed**

- `lerobot_train.py` and `rl/learner.py` call `make_tracker(cfg)` instead of constructing `WandBLogger`, and type against the protocol.
- **HF Jobs**: the pod's `WANDB_API_KEY` is decided by the resolved tracker, not `cfg.wandb.enable`, so `--tracker.type=wandb` arrives authenticated exactly like `--wandb.enable=true`. A null `tracker` is stripped from the staged config, as `save_checkpoint_to_hub` already is, keeping older `lerobot-gpu` images happy.
- **Docs**: `installation.mdx` gains the extra plus an Experiment Tracking section with one command per tracker; `cheat-sheet.mdx` gains the tracker options.

**Two trackio-specific details worth knowing**

- **Run identity:** trackio keys runs by name and resolves a resume to the _most recently written_ run of that name. A fresh run is therefore named after its output dir (`15-30-14_pusht_act`), not `job_name`, which repeats across runs of the same policy and env. The name is stored on the config so resume reattaches to the right run.
- **Durability on HF Jobs:** trackio's DB lives under `$HF_HOME`, which the pod discards on exit. `submit_to_hf` refuses a remote trackio run with no `space_id` rather than letting it train and drop every metric.

## How was this tested (or how to run locally)

- **New tests:** `tests/test_tracker.py` (22) and 3 in `tests/jobs/test_hf.py`.

  ```bash
  pytest -q tests/test_tracker.py tests/jobs/test_hf.py
  ```

  Covering: selection via both flag spellings; that a selected tracker cannot report itself disabled and `--tracker.enable` is rejected; the `train_config.json` round-trip **resume depends on**, for each tracker and for none; the legacy `--wandb.*` path including a config written before `tracker` existed; `make_tracker` dispatch to each logger; and `TrackioLogger`'s run naming, resume reattachment, metric filtering, custom step key, video and artifact paths. **Both loggers run against stubbed `wandb` / `trackio` modules, so the suite passes with neither extra installed.**

- **Quality:** `pre-commit run --all-files` → **all 19 hooks pass** (ruff, ruff-format, mypy, bandit, gitleaks, typos, pyupgrade, prettier, zizmor).
- **Install:** `uv sync --locked` verified against each CI tier (`test`, `+dataset`, `+hardware`, `+viz`) plus `+trackio`, with uv 0.11.30.
- **Manual:** trained with `--tracker.type=trackio` and viewed the run in `trackio show`; both documented commands run through `validate()`.

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [x] All tests pass locally (`pytest`)
- [x] Documentation updated
- [ ] CI is green
- [ ] Community Review: I have reviewed another contributor's open PR and linked it here: # (insert PR number/link)

## Reviewer notes

1. **Should the `--wandb.*` flags be deprecated?** My patch keeps them working and silently builds
   the equivalent tracker, with no deprecation warning, because deprecating a documented flag feels
   like a call for the team rather than something to settle inside a feature PR. Happy to add the
   warning and update the remaining docs if you would rather move users over.
2. **`WandBConfig.enable` has no meaning once a tracker is selected**, since selecting one is what
   enables it (and `--tracker=null` turns tracking off again). The wandb tracker config inherits the
   field from the existing `WandBConfig` and I neutralize it rather than duplicating the field list;
   there is a cleaner shape that costs one extra dataclass, and I am happy to go either way.

### CODE REVIEW

**Comment by zuoxingdong** [comment]
@imstevenpmwork as a follow-up, just resolved some conflicts with the latest main branch
